addons/l10n_es_edi_tbai/models/account_move.py

```diff
@@ -0,0 +1,285 @@
+    )
+    l10n_es_tbai_cancel_xml_id = fields.Many2one(
+        comodel_name="ir.attachment", readonly=True, copy=False,
+        string="Cancellation XML",
+        help="Cancellation XML sent to TicketBAI. Kept if accepted or no response (timeout), cleared otherwise.",
+    )
+
+    # Non-stored fields
+    l10n_es_tbai_id = fields.Char(string="TicketBAI ID", compute="_compute_l10n_es_tbai_id")
+    l10n_es_tbai_sequence = fields.Char(string="TicketBAI sequence", compute="_compute_l10n_es_tbai_seq_num")
+    l10n_es_tbai_number = fields.Char(string="TicketBAI number", compute="_compute_l10n_es_tbai_seq_num")
+    l10n_es_tbai_qr = fields.Char(string="TicketBAI QR code URL", compute="_compute_l10n_es_tbai_qr")
+    l10n_es_tbai_qr_escaped = fields.Char(string="TicketBAI QR code, escaped", compute="_compute_l10n_es_tbai_qr")
+    l10n_es_tbai_signature = fields.Char(string="TicketBAI XML signature", compute="_compute_l10n_es_tbai_values")
+    l10n_es_tbai_registration_date = fields.Date(
+        string="TicketBAI registration date",
+        help="Technical field to keep the date the invoice was sent the first time as the date the invoice was "
+             "registered into the TicketBai system.",
+        compute="_compute_l10n_es_tbai_values",
+    )
+
+    # Optional fields
+    l10n_es_tbai_refund_reason = fields.Selection(
+        selection=[
+            ('R1', "R1: Art. 80.1, 80.2, 80.6 and rights founded error"),
+            ('R2', "R2: Art. 80.3"),
+            ('R3', "R3: Art. 80.4"),
+            ('R4', "R4: Art. 80 - other"),
+            ('R5', "R5: Factura rectificativa en facturas simplificadas"),
+        ],
+        string="Invoice Refund Reason Code (TicketBai)",
+        help="BOE-A-1992-28740. Ley 37/1992, de 28 de diciembre, del Impuesto sobre el "
+        "Valor Añadido. Artículo 80. Modificación de la base imponible.",
+        copy=False,
+    )
+
+    # -------------------------------------------------------------------------
+    # COMPUTE & API-DECORATED METHODS
+    # -------------------------------------------------------------------------
+
+    @api.depends('move_type', 'company_id')
+    def _compute_l10n_es_tbai_is_required(self):
+        for move in self:
+            move.l10n_es_tbai_is_required = move.is_sale_document() \
+                and move.country_code == 'ES' \
+                and move.company_id.l10n_es_tbai_tax_agency
+
+    @api.depends('l10n_es_tbai_is_required', 'edi_document_ids.state')
+    def _compute_l10n_es_tbai_is_in_chain(self):
+        """
+        True iff invoice has been posted to the chain and confirmed by govt.
+        Note that cancelled invoices remain part of the chain.
+        """
+        for move in self:
+            tbai_doc_ids = move.edi_document_ids.filtered(lambda d: d.edi_format_id.code == 'es_tbai')
+            move.l10n_es_tbai_is_in_chain = move.l10n_es_tbai_is_required \
```

Comment:
The chain index is set before posting the document, in case a timeout prevents us from knowing if it's been accepted or not.
is_in_chain means it's positively been confirmed to be in the chain by the govt.

Reply:
It could still be a method instead of a field then...

Reply:
There are some dependencies on that field (and the post_xml_id)...

Reply:
The dependencies do not seem strictly necessary, changed to a method.